rust/stackable-cockpitd/src/handlers/ui.rs

```diff
@@ -16,16 +16,20 @@ pub fn router() -> Router {
 async fn ui() -> Html<&'static str> {
     Html(stackable_cockpit_web::INDEX_HTML)
 }
-async fn asset(Path(name): Path<String>) -> impl IntoResponse {
-    (
-        AppendHeaders([(
-            CONTENT_TYPE,
-            match name.split_once('.') {
-                Some((_, "js")) => HeaderValue::from_static("text/javascript"),
-                Some((_, "css")) => HeaderValue::from_static("text/css"),
-                _ => HeaderValue::from_static("application/octet-stream"),
-            },
-        )]),
-        stackable_cockpit_web::ASSETS[&name],
-    )
+
+/// Adds (or replaces) the Content-Type header with the type of the served asset.
+/// So far only javascript and css are supported, for all the other types
+/// `application/octet-stream` will be used.
+async fn asset(mut headers: HeaderMap, Path(name): Path<String>) -> impl IntoResponse {
+    headers.insert(
```

Comment:
This doesn't do what you want it to. The `HeaderMap` argument isn't some base set of headers defined by the framework, but the headers sent by the client. What this PR actually does is to reflect any headers sent by the user back at them.

Reply:
You can test this yourself: `curl 'http://localhost:8000/ui/assets/index-1272d2e9.js' -H 'foo: bar' -D- -o /dev/null | grep foo`.